src/common/container.h

```diff
@@ -93,6 +93,24 @@ static inline void smartlist_swap(smartlist_t *sl, int idx1, int idx2)
   }
 }
 
+/** Return true if there is shallow equality between smartlists -
+ * i.e. all indices correspond to exactly same object (pointer
+ * values are matching). Otherwise, return false.
+ */
+static inline int smartlist_shallow_eq(const smartlist_t *s1,
+                                       const smartlist_t *s2)
+{
```

Comment:
Since we opted for a generic function here we should also check that `s1` and `s2` are non-NULL and `return 0` if any of them are NULL.

Reply:
Introduced some extra checks in cda8adc876b87443859577e54f080f56b8f5e583.

Reply:
LGTM